test/vim_interface.py

```diff
@@ -85,18 +86,25 @@ def vim_executable(self):
     def has_version(self, major, minor, patchlevel):
         if self._version is None:
             output = subprocess.check_output([self._vim_executable, "--version"])
-
-            _major = 0
-            _minor = 0
-            _patch = 0
             for line in output.decode("utf-8").split("\n"):
                 if line.startswith("VIM - Vi IMproved"):
-                    _major, _minor = map(int, line.split()[4].split("."))
+                    self._version = tuple(map(int, line.split()[4].split(".")))
                 if line.startswith("Included patches:"):
-                    _patch = int(line.split(":")[-1].strip().split("-")[-1])
-            self._version = (_major, _minor, _patch)
-
-        return self._version >= (major, minor, patchlevel)
+                    for chain in line.split(":")[-1].split(","):
+                        if "-" in chain:
+                            lower, higher = map(int, chain.split("-"))
+                            self._patches.update(range(lower, higher))
+                        else:
+                            self._patches.add(int(chain))
+                if line.startswith("Extra patches:"):
+                    # Vim before 9.0 also prints extra pacthes as full version
+                    # numbers. Example: 8.2.3402, 8.2.3403, 8.2.3409, 8.2.3428
```

Comment:
This is not a change in behavior.  There are two different ways that Vim tracks patches.

1. [Included patches](https://github.com/vim/vim/blob/b40a2fb3b3e4fef137eebb3bab9c02adc5540658/src/version.c#L696) are the different patch levels for the current version that are included.
2. [Extra patches](https://github.com/vim/vim/blob/b40a2fb3b3e4fef137eebb3bab9c02adc5540658/src/version.c#L3090-L3097) are free-form strings that aren't usually used in official Vim builds, but can be used in other builds.  For Debian, that's where I document patches that are back ported from a different release (e.g, applying 8.2.3402 to an 8.1 build of Vim).

Since there's no defined structure for extra patches, I'm not sure it's worth parsing.